v1.15 Backports 2024-03-06

giorio94 opened the pull request:
* [x] #30928 (@giorio94)
 * [x] #30916 (@giorio94)
     - :information_source: Skipped the renovate-related commit, as not relevant in stable branches.
 * [x] #31198 (@giorio94)

The last commit additionally switches the k8s version used for Ginkgo tests from `quay.io/cilium/kindest-node:v1.29.0-rc.1` to `kindest/node:v1.29.2`.

Once this PR is merged, a GitHub action will update the labels of these PRs:
```upstream-prs
 30928 30916 31198
```

giorio94 committed:
gha: centralize kind version and image definition in set-env-variables

[ upstream commit 394b3de26a4e2235ec25399861e12886b507f335 ]

Let's define kind-related variables (i.e., version, k8s image and k8s
version) inside the set-env-variables action. One all consumers will
have been migrated through the subsequent commit, this will ensure
consistency across workflows, simplify version bumps as well as the
introduction of new workflows depending on them. One extra byproduct
is that renovate updates will also stop requesting reviews from all
the different teams owning each specific workflow.

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>
Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
gha: migrate workflows to use the global kind-related variables

[ upstream commit aabdfa73d3d83edda3935277bd08c4c4c0bf5b68 ]

Let's switch all the workflows over to using the globally defined
kind-related variables, and remove the workflow specific definitions.
This also addresses a few cases which didn't specify any version.

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>
Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
gha: drop unused check_url environment variable

[ upstream commit e17cf21d9720493766c9f1d12c2d75c842f26e86 ]

This variable used to be used in combination with the Sibz/github-status-action
action, which we replaced with myrotvorets/set-commit-status-action when
reworking the workflows to be triggered by Ariane [1]. Given it is now
unused, let's get rid of the leftover environment variable, so that we
also stop copying it to new workflows.

[1]: 9949c5a1891a ("ci: rework workflows to be triggered by Ariane")

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>
Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
gha: don't wait for kind clusters to become ready

[ upstream commit 39637d6d2385baab556078d844f431522d99f616 ]

They will never, because no CNI is present at that point. Hence, let's
just avoid wasting one minute waiting for the timeout to expire.

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>
Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 commented on .github/actions/set-env-variables/action.yml:
```diff
@@ -16,3 +16,13 @@ runs:
+        # renovate: datasource=github-releases depName=kubernetes-sigs/kind
+        KIND_VERSION="v0.22.0"
+        # renovate: datasource=docker
+        KIND_K8S_IMAGE="quay.io/cilium/kindest-node:v1.29.0-rc.1@sha256:6631d58da3569930cf21697c2113feb9c76bc044c1c13d98808a3695dd91d8b0"
```
@aanm Should we bump the k8s image version to v1.29.2? I've currently kept the one previously used in v1.15, but it looks strange to me that we test a release candidate k8s version in a stable Cilium branch.

aanm replied:
Yeah, this should be `kindest/node:v1.29.2@sha256:51a1434a5397193442f0be2a297b488b6c919ce8a3931be0ce822606ea5ca245`

giorio94 replied:
Updated

giorio94 committed:
gha: checkout target branch instead of the default one

[ upstream commit 6716a9c01b69d88da0c3316fe8b3640180bbafb1 ]

Currently, the GHA workflows running tests triggered on pull_request
and/or push events initially checkout the default branch to configure
the environment variables, before retrieving the PR head. However,
this is problematic on stable branches, as we then end up using the
variables from the default (i.e., main) branch (e.g., Kubernetes
version, Cilium CLI version), which may not be appropriate here.

Hence, let's change the initial checkout to retrieve the target (i.e.,
base) branch, falling back to the commit in case of push events.
This ensure that we retrieve the variables from the correct branch,
and matches the behavior of Ariane triggered workflows.

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>

giorio94 committed:
gha: switch kind images back to kind

Kind has released stable versions for k8s 1.29 so we can use this
image instead of the cilium kindest for ginkgo tests. The same
version has already been configured for the rest of the workflows
in the previous commits.

Signed-off-by: Marco Iorio <marco.iorio@isovalent.com>
